### ISSUE TITLE
S3459 flags unassigned fields in serializable classes

### ISSUE THREAD

**Issue by Nezz** [SonarSource/sonar-dotnet#5451]
### Description

S3459 flags unassigned fields in serializable classes

### Repro steps
Analyze the following code:
```
[Serializable]
class Foo
{
    private int Bar;
    
    public void Print()
    {
        Console.WriteLine(Bar);
    }
}
```

### Expected behavior

Fields in classes and structs annotated with SerializableAttribute should not trigger S3459.

### Actual behavior

The following code smell is generated:
```
Remove unassigned field 'Bar', or set its value.
```

### Known workarounds

No known workaround

### Related information

* C#/VB.NET Plugins version: N/A (using GitHub Actions)
* Visual Studio version: N/A
* MSBuild / dotnet version: dotnet 5 and 6
* SonarScanner for .NET version (if used): Latest version used by GitHub Actions
* Operating System: Linux, Windows, macOS

**Comment by costin-zaharia-sonarsource** [comment]
Hi @Nezz , thanks for reporting this. 

I've updated your reproduction code a bit since the rule considers only used private fields.

**Comment by Nezz** [comment]
@costin-zaharia-sonarsource Here's what we see:

<img width="622" alt="Screenshot 2022-03-14 at 12 42 30" src="https://user-images.githubusercontent.com/431167/158156628-30b4d20c-3f56-4ae8-9f95-17eb7be7032c.png">
<img width="503" alt="Screenshot 2022-03-14 at 12 42 39" src="https://user-images.githubusercontent.com/431167/158156635-6ce01f85-978f-4263-b949-e5d3c011f9cd.png">

**Comment by costin-zaharia-sonarsource** [comment]
Thanks, @Nezz.

I understand now the full context. The field is in a private class. This is relevant since the rule checks only private fields or fields from private classes.
